crates/bevy_app/src/app.rs

```diff
@@ -87,6 +87,9 @@ pub struct App {
     /// [`WinitPlugin`]: https://docs.rs/bevy/latest/bevy/winit/struct.WinitPlugin.html
     /// [`ScheduleRunnerPlugin`]: https://docs.rs/bevy/latest/bevy/app/struct.ScheduleRunnerPlugin.html
     pub(crate) runner: RunnerFn,
+    /// The list of hooks to run distributing exclusive app access to outside code.
+    pub(crate) hooks: HashMap<usize, RunnerHookFn>,
+    pub(crate) hook_id_count: usize,
```

Comment:
hook_id_count is never decremented, so this isn't the "current count of hooks" if a hook is removed which feels like it could be confusing if not documented.

```suggestion
    /// The next id that will be used for a newly registered hook.
    /// This value does not track the currently active hook amount as it is never decremented.
    pub(crate) hook_id_count: usize,
```